README.md
[logo]: https://github.com/mcc85s/FightingEntropy/blob/main/Graphics/OEMlogo.bmp

<p align="center" width="100%">
    <img width="66%" src="https://github.com/mcc85s/FightingEntropy/blob/main/Graphics/banner.png">
</p>

| Logo    | Company                        | Project                  | Objective                                           |
|:--------|:-------------------------------|:-------------------------|:----------------------------------------------------|
| ![logo] | **Secure Digits Plus (π) LLC** | **[FightingEntropy(π)]** | Beginning the fight against ID theft and cybercrime |

# About
**[FightingEntropy(π)]** is a modification for **Windows PowerShell** that is meant for various tasks related to:
- system administration
- networking
- virtualization
- security
- graphic design
- system management/maintenance

# Demo
| Date     | Name                                   | Url                          |
|:---------|:---------------------------------------|:-----------------------------|
| 10/28/22 | **[FightingEntropy(π)][2022.10.1]**    | https://youtu.be/S7k4lZdPE-I |

This module is rather _experimental_ and incorporates a lot of moving parts, so it has many areas of development.
The end goal of this module, is to provide protection against:
- identity theft
- cybercriminals
- douchebags
- malware
- viruses
- ransomware
- hackers who have malicious intent

Many of the tools in the wild are able to be circumvented by some of these hackers and cybercriminals.
If you don't believe me...? That's fine. That's why [this link to a particular website about a particular event](https://en.wikipedia.org/wiki/2020_United_States_federal_government_data_breach), exists.

**[FightingEntropy(π)]** is meant to extend many of the capabilities that come with Windows.

# Current (Alpha)
| Version       | Date                  | Guid                                   |
|:--------------|:----------------------|:---------------------------------------|
| **2022.12.0** | `12/14/2022 14:26:18` | `5e6c9634-1c88-49a2-8794-2970095d8793` |

# Prerequisites
1) A system running [**Windows PowerShell**] on: 
- [**Windows 10/11**]
- [**Windows Server 2016/2019/2021**]
   
2) [**Execution Policy**] must be set to [**bypass**]

3) Must be running a [**PowerShell**] session with [**administrative privileges**]

# Installation
1) [**Load the module into memory**], which can be done be using this command:
   
`irm https://github.com/mcc85s/FightingEntropy/blob/main/FightingEntropy.ps1?raw=true | iex`

...or just (copying + pasting) the content of the file...

`https://github.com/mcc85s/FightingEntropy/blob/main/FightingEntropy.ps1`

...into the [**PowerShell**] session, and pressing [**&lt;Enter&gt;**]... and then [**boom**].
You're rollin'.

2) Once the [**module is loaded into memory**], enter the following:
____________________________________
| Operation  | Instructions        |
|:-----------|:--------------------|
| Install    | `$Module.Install()` | 
| Remove     | `$Module.Remove()`  |
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯

# Author
| Name                | Motto                                          | Contact                    | Resume           |
|:--------------------|:-----------------------------------------------|:---------------------------|:-----------------|
| Michael C. Cook Sr. | Sometimes you gotta take the throttle in life. | securedigitsplus@gmail.com | [Short](https://github.com/mcc85s/FightingEntropy/blob/main/Docs/2022_1010-(MCC%20Short%20Resume).pdf)/[Full]( https://github.com/mcc85s/FightingEntropy/blob/main/Docs/2022_1010-(MCC%20Full%20Resume).pdf) |

# Videos
These various videos demonstrate some portion of [FightingEntropy(π)]
| Index | Date    | Name                                      | Url                          |
|:------|:--------|:------------------------------------------|:-----------------------------|
|     0 | 01/2023 | PowerShell Virtualization Lab + FEDCPromo | https://youtu.be/9v7uJHF-cGQ |
|     1 | 12/2021 | [FightingEntropy(π)][FEInfrastructure]    | https://youtu.be/6yQr06_rA4I |
|     2 | 10/2021 | Advanced Domain Controller Promotion      | https://youtu.be/O8A2PDfQOBs |
|     3 | 03/2021 | A Deep Dive: PowerShell and XAML          | https://youtu.be/NK4NuQrraCI |
|     4 | 05/2022 | Wireless Network Scanning Utility         | https://youtu.be/35EabWfh8dQ |
|     5 | 09/2021 | PowerShell Deployment FE Wizard           | https://youtu.be/lZX5fAgczz0 |
|     6 | 01/2019 | 2019_0125-(Computer Answers - MDT)        | https://youtu.be/5Cyp3pqIMRs |
|     7 | 06/2021 | Install-pfSense                           | https://youtu.be/E_uFbzS0blQ |
|     8 | 06/2021 | Advanced System Administration Lab        | https://youtu.be/xgffIccX1eg |
|     9 | 06/2021 | Windows Image Extraction                  | https://youtu.be/G10EuwlNAyo |
|    10 | 08/2021 | Flight Test Part 1                        | https://drive.google.com/file/d/1qdS_UVcLTsxHFCpuwK16NQs0xJL7fv0W |
|    11 | 08/2021 | Flight Test Part 2                        | https://youtu.be/vg359UlYVp8 |
|    12 | 09/2021 | cimdb                                     | https://youtu.be/vA8_HLZ--mQ |
|    13 | 05/2019 | Hybrid - Desired State Controller         | https://youtu.be/C8NYaaqJAlI |
|    14 | 11/2019 | Methodologies                             | https://youtu.be/bZuSgBK36CE |
|    15 | 08/2019 | Education/Exhibition Program Design       | https://youtu.be/v6RrrzR5v2E |


Theme/changelog.md
# Change Log
All notable changes to the "bossmode" extension will be documented in this file.
- Initial release


Video/sdp.ps1
$Ctrl = New-YouTubePortfolio -Name "Michael C. Cook Sr." -Company "Secure Digits Plus LLC"

("Secure Digits Plus LLC","securedigitsplus@gmail.com","securedigitsplusllc3084"),
("Michael C. Cook","michael.c.cook.85@gmail.com","michaelcook7389"),
("Michael Cook","mykalcook@gmail.com","mykalcook"),
("Secure Digits Plus","sdp12065@gmail.com","securedigitsplus2892") | % { 

    $Ctrl.Add($_[0],$_[1],$_[2])
}

# 0 / [Secure Digits Plus LLC (securedigitsplusllc3084)]
("02-19-2023","01:51:25","u3y4NHD_svUz","Russia - Crushing Defeat"),
("02-18-2023","01:28:24","CPJmWpzEDBE","Wrongful Robert"),
("02-17-2023","00:11:53","1s-qefkOjXQ","Malta Tech Park"),
("02-15-2023","04:41:01","p4MPjtd2vpw","Halfmoon Historic"),
("02-14-2023","00:44:09","Ev97ifWx-L8","Hive Mentality"),
("02-13-2023","02:12:00","MI2imvb-4Vg","Invoke-Cimdb"),
("02-12-2023","00:38:09","ENpIkuIbqZU","NYSP Mafiosos"),
("02-11-2023","00:20:17","SGKypOeZuX4","Build-Discography"),
("02-10-2023","00:06:24","Ua1r_7o5BpI","Weeping Angel"),
("02-10-2023","00:06:40","4V6Wwzxlqzw","Track 20 (Original Music)"),
("02-10-2023","00:06:31","23d9VPFr0u4","Futile Intent (Original Music)"),
("02-09-2023","01:14:32","sqSMw4JzQoM","New-YouTubeProfile"),
("02-07-2023","00:17:25","zIDnV1BsiCA","rastructure"),
("02-07-2023","01:59:05","Dj9E-eNe4Tg","Golub Corp. - Empire State Plaza"),
("02-04-2023","00:27:53","aejBB77d_oA","Why William Moak is stupid"),
("02-04-2023","03:04:48","j58BO1p_EJ8","Write-Element (Demonstration)"),
("02-01-2023","02:01:57","nqTOmNIilxw","New-VmController [Flight Test v2.0] Part I"),
("01-12-2023","01:14:43","9v7uJHF-cGQ","PowerShell | Virtualization Lab + FEDCPromo"),
("11-22-2022","01:15:56","Y7wEiuNJhN0","Search-WirelessNetwork"),
("10-28-2022","01:22:26","S7k4lZdPE-I","[FightingEntropy()][2022.10.1]"),
("10-15-2022","00:13:33","-GScIS_PlOo","Intellectual Property Theft"),
("10-14-2022","00:15:03","ZiTVgtg68Jc","Indicative Behaviors"),
("10-04-2022","00:00:20","dU_5rdVkCD8","God Mode Cursor"),
("10-03-2022","00:31:32","xhMQbOoDvOc","Behavioral Analysis"),
("09-29-2022","00:02:15","K738wZSKcjo","Write Progress Extension"),
("09-26-2022","00:01:12","tW80Zj_H6Fw","God Mode Cursor"),
("09-27-2022","00:57:04","1ABQ6rfRg8Y","Correlations"),
("09-08-2022","02:11:03","z6_GeVYbcC4","Developing a GUI with style"),
("08-11-2022","00:05:18","QSuge7p5_I8","PowerShell Resume"),
("08-03-2022","00:08:43","vY2fIhS9ruo","Scribbles locking ANY word editor"),
("04-15-2022","09:52:39","CCOTI6_Veoo","Event Logs Utility (Work in progress)"),
("04-05-2022","00:05:18","35EabWfh8dQ","Wireless Network Scanner Utility (PowerShell/Xaml)"),
("02-15-2022","00:20:27","e4VnZObiez8","A Matter of National Security"),
("12-05-2021","01:49:57","6yQr06_rA4I","[FightingEntropy(π)] FEInfrastructure Preview/Demo"),
("10-25-2021","01:56:00","1E-3POI29Jo","Ambitious Automation"),
("10-14-2021","00:38:23","4q3fWhTOuFk","DCPromo and Networking"),
("10-14-2021","00:09:32","AdAilZe1EJ4","FEInfrastructure (Alpha/Beta)"),
("10-02-2021","02:43:44","vbAH4l6Bm7A","Cross Platform Development"),
("09-23-2021","00:08:17","lZX5fAgczz0","[FightingEntropy(π)]://FEWizard PowerShell Deployment feat. DVR"),
("09-16-2021","04:05:38","m84TElKgAFg","PowerShell Deployment Wizard (featuring DVR) [WIP]"),
("09-10-2021","00:12:25","_i2bPQT1ZtU","Resolve-LogAddressList"),
("05-23-2020","01:35:44","HT4p28bRhqc","Virtual Tour"),
("01-25-2019","00:32:39","5Cyp3pqIMRs","Computer Answers - MDT") | % {

    $Ctrl.AddVideo(0,$_[0],$_[1],$_[2],$_[3])
}

# 1 / [Michael C. Cook (michaelcook7389)]
("06-29-2022","00:00:19","W0SZ9Iby3VY","SCSO and NYSP trying to be sneaky"),
("08-30-2021","02:33:21","vg359UlYVp8","FEDeploymentShare Part 2"),
("08-11-2021","00:10:44","70NJcRFTZo8","New-FEDeploymentShare"),
("07-26-2021","00:38:48","eaAyCJ7zsZw","FEDeploymentShare (Part 1)"),
("07-26-2021","01:13:13","jmvzDV4RCQY","2021 07 26 12 24 59"),
("06-06-2021","01:17:43","9zjQqQ0sKPg","[FightingEntropy(π)][(2021.6.0)]@[FEModule GUI (Part 2)]"),
("06-06-2021","02:42:22","Zc8Fq4E3Nc0","[FightingEntropy(π)][(2021.6.0)]@[FEModule GUI (Part 1)]"),
("03-09-2021","00:57:16","NK4NuQrraCI","A Deep Dive: Demonstration - Building XAML/WPF via PowerShell"),
("02-15-2021","00:53:26","45U1qFlU_74","FightingEntropy (π) Testing..."),
("12-13-2020","00:02:15","SmsqBv200B8","About 'My Briefcase' from (Windows 95/98)"),
("12-12-2020","01:05:26","51Ymdd9XqQ","PowerShell XAML/WPF Classes"),
("11-07-2020","00:11:40","RZ1SEUwSb9Q","FightingEntropy - New Deployment Share GUI"),
("11-05-2020","00:06:45","KI0-V5OGy5w","FightingEntropy - (Windows Image Extraction)"),
("10-14-2020","00:21:57","MRcLWQCJwt4","Drafting and Design meets Programming"),
("09-30-2020","00:15:11","bOpZIeQF5KQ","NMap Vendor List"),
("09-16-2020","00:16:00","dBpfRazxAi0","Write-Theme (Update/[FEObject])"),
("09-05-2020","01:25:34","6SubeX4gvD0","(Cruise Control) 2/2"),
("09-05-2020","00:56:36","afLnNLdG2PU","(Cruise Control) 1/2"),
("09-03-2020","00:11:05","nFijaWGytnA","FightingEntropy [FEModule/FERoot]"),
("08-13-2020","00:10:17","QGtzvsGSUak","Write-Theme & Show-ToastNotification (Source Code/Lesson Plan/Resume in desc.)"),
("08-09-2020","00:32:35","EQXRrPQ-iOw","PowerShell/C# Classes"),
("05-21-2020","00:29:25","hbBVNerI_Qk","Service Configuration Tool"),
("11-28-2019","00:43:48","bZuSgBK36CE","Some Powershell scripting/programming methodologies"),
("08-28-2019","01:47:16","v6RrrzR5v2E","Education/Exhibition of Programming/Design/Engineering with Powershell/C#/.Net"),
("05-28-2019","01:02:43","C8NYaaqJAlI","Hybrid | Desired State Controller | Process Exhibition (The long version)"),
("03-24-2019","00:47:24","qiZcHqkAzbs","Hybrid(Server+Client)"),
("08-19-2018","00:02:00","bPdWt7kcd3M","News 10 - Smart TV report"),
("01-16-2018","00:23:55","pi1DIQWuce8","Driving in Snow (feat. Audi A4 Quattro)") | % { 

    $Ctrl.AddVideo(1,$_[0],$_[1],$_[2],$_[3])
}

# 2 / [Michael Cook (mykalcook)]
("08-13-2022","00:41:08","eD0-VQ2y_yg","2018_1220-(OPNsense Configuration)"),
("09-08-2021","00:05:52","vA8_HLZ--mQ","cimdb2"),
("06-05-2021","00:05:16","xN53K9oGCME","[Q3A] 20KDM2 - Return to Castle: Quake (2002)"),
("06-05-2021","00:10:33","dyHwm9AdkQs","[Q3A] 20KDM1 - Tempered Graveyard (2001)"),
("06-05-2021","00:16:16","rwyHCNnwlkM","[Q3A] 20KCTF1 - Out Of My Head (2002)"),
("06-05-2021","00:11:25","EG8UyJSMK3Y","[Q3A] 20KDM3 - Insane Products (2005)"),
("02-17-2021","00:01:02","ApEqOjAoAfM","2017_0831-(Threadripper Gen1)"),
("10-23-2020","01:08:20","iOKOkJJ1ZbQ","CentOS 7 Setup"),
("10-20-2020","00:56:33","YDWm-f7WEWs","2019_1004-AsusQ504UA301(tilted)"),
("10-20-2020","00:41:08","eYEESvPOWh4","Configuring pfSense/Hardened BSD"),
("10-20-2020","00:01:38","0nEiGijjOEY","Network Troubleshooting"),
("10-20-2020","00:03:34","LfZW-s0BMow","Spectrum Cable Modem Reset"),
("05-16-2019","00:41:01","jplOHy_b1bA","Secure Digits Plus LLC @ Hybrid | Desired State Controller ( Beta Preview )"),
("03-02-2019","01:01:30","RypW9xbClJo","2019 03 01 23 35 05"),
("07-28-2018","00:09:55","joioy6bIyPE","Trickshot - Wicked") | % {

    $Ctrl.AddVideo(2,$_[0],$_[1],$_[2],$_[3])
}

# 3 / [Secure Digits Plus (sdp12065)]
("10-20-2021","00:17:10","O8A2PDfQOBs","[FightingEntropy]://Domain Controller Promotion"),
("06-30-2021","00:10:09","G10EuwlNAyo","PowerShell - (Xaml/WPF GUI) Windows Image Extraction"),
("06-27-2021","00:40:28","xgffIccX1eg","PowerShell - Advanced System Administration Lab"),
("06-20-2021","00:10:03","E_uFbzS0blQ","Install-pfSense (Auto)"),
("06-06-2021","00:03:58","acYldTcKUAs","[FightingEntropy(π)][(2021.6.0)] Installation + FEModule GUI"),
("06-04-2021","00:07:05","UYnCj5Hcq6o","[FightingEntropy(π)][(2021.6.0)]"),
("03-16-2021","01:38:47","4yFKeK9CzU0","FightingEntropy (π) v2021.3.1 | New-FEDeploymentShare GUI") | % { 

    $Ctrl.AddVideo(3,$_[0],$_[1],$_[2],$_[3])
}
